src/sentry/tasks/assemble.py

```diff
@@ -311,14 +326,91 @@ def _create_artifact_bundle(
                     artifact_bundle=artifact_bundle,
                     source_file_type=source_file_type.value,
                 )
+        else:
+            raise AssembleArtifactsError(
+                "uploading a bundle without debug ids or release is prohibited"
+            )
+
+
+def get_with_manifest_fallback(value, manifest_key, manifest, throw=True):
+    if value is None:
+        manifest_value = manifest.get(manifest_key)
+        if manifest_value is None and throw:
+            raise AssembleArtifactsError(f"failed to load {manifest_key} from the manifest")
+
+        return manifest_value
+
+    return value
+
+
+def handle_assemble_for_release_file(bundle, archive, manifest, organization, version):
+    org_slug = manifest.get("org")
+    if organization.slug != org_slug:
+        raise AssembleArtifactsError("organization does not match uploaded bundle")
+
+    release_name = manifest.get("release")
+    if release_name != version:
+        raise AssembleArtifactsError("release does not match uploaded bundle")
+
+    try:
+        release = Release.objects.get(organization_id=organization.id, version=release_name)
+    except Release.DoesNotExist:
+        raise AssembleArtifactsError("release does not exist")
 
-            return True
+    dist_name = manifest.get("dist")
+    dist = None
+    if release and dist_name:
```

Comment:
It's impossible for `release` to not be defined at this stage due to `except Release.DoesNotExist` so you can drop this check and write

```
dist = release.add_dist(dist_name) if dist_name else None
```

Reply:
Overlooked that during refactor, thanks!